stuffer/s2n_stuffer_base64.c

```diff
@@ -140,18 +143,21 @@ int s2n_stuffer_read_base64(struct s2n_stuffer *stuffer, struct s2n_stuffer *out
             *ptr = ((value3 << 6) & 0xc0) | (value4 & 0x3f);
             ptr++;
         }
-
     } while (bytes_this_round == 3);
 
-    return 0;
+    return S2N_SUCCESS;
 }
 
 int s2n_stuffer_write_base64(struct s2n_stuffer *stuffer, struct s2n_stuffer *in)
 {
-    uint8_t outpad[4];
-    uint8_t inpad[3];
-    struct s2n_blob o = {.data = outpad,.size = sizeof(outpad) };
-    struct s2n_blob i = {.data = inpad,.size = sizeof(inpad) };
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(in));
+    uint8_t outpad[4] = {0};
+    uint8_t inpad[3] = {0};
+    struct s2n_blob o = {0};
+    GUARD(s2n_blob_init(&o, outpad, sizeof(outpad)));
+    struct s2n_blob i = {0};
+    GUARD(s2n_blob_init(&i, inpad, sizeof(inpad)));
```

Comment:
```suggestion
    s2n_stack_blob(o, 4, 4);
    s2n_stack_blob(i, 3, 3);
```

Reply:
What about `outpad` and `inpad`? We still need to add it afterwards
```c
o.data = outpad;
i.data = inpad;
```

Reply:
nope, https://github.com/awslabs/s2n/blob/c1cf61787b92884ff9e476e56154f73785415ae7/utils/s2n_blob.h#L49

Reply:
Implemented as suggested.